core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala

```diff
@@ -163,6 +163,26 @@ abstract class QuorumTestHarness extends Logging {
   // That way you control the initialization order.
   @BeforeEach
   def setUp(testInfo: TestInfo): Unit = {
+    Exit.setExitProcedure((code, message) => {
+      try {
+        throw new RuntimeException(s"exit(${code}, ${message}) called!")
+      } catch {
+        case e: Throwable => error("test error", e)
+          throw e
```

Comment:
Did you mean to add this exception handling? Are you trying to log the exception before throwing it? If so, how about:
```suggestion
        val e = new RuntimeException(s"exit(${code}, ${message}) called!")
        error("Throwing an exception because the test called exit", e)
        throw e
```

Similarly for the halt override.

Reply:
the issue with doing that is that you will not get a stack trace that way (I think)

Reply:
It is close enough. Java constructs the stacktrace when the exception is constructed. Line 168 in this case. I am okay either way.

Reply:
Yeah, I think you're right. I guess this way of doing it seems a bit clearer to me, though. I do want the message logged before the cleanup call is made as well.